### DISCUSSION TITLE
[✨] DOM in the server

### DISCUSSION THREAD

**Discussion by GrandSchtroumpf** [QwikDev/qwik-evolution#149] [Proposals For Qwik]
### Is your feature request related to a problem?

I'm building a select component that displays the text content of the selected option's label inside a button element : 
```jsx
const origin = useSignal<HTMLElement>(); // Root of the select
const name = 'select';
const displayText = useSignal('');
const update = $(() => {
  // In the server origin.value is a MockElement, but has no querySelectorAll attached to it
  const inputs = origin.value!.querySelectorAll<HTMLInputElement>(`input[name="${name}"]`);
  for (const input of inputs) {
    if (input.checked && input.value !== 'on') {
      const text = Array.from(input.labels ?? []).map(label => label.textContent).join(' ');
      value.push(text);
    }
  }
  displayText = value.join(', ');
});

const initialValue = useFormValue(name); // Get the initialValue from the form if any
useTasks$(() => {
  if (initialValue) update(); // Update the display text if there is an initialValue
})
```

### Describe the solution you'd like

Currently `querySelectorAll` is not a function in the server.
It would be nice to be able to manipulate the DOM in the server too. I know libs like domino enable that.
Maybe `qwik-dom` would help with that too.

### Describe alternatives you've considered

I use `useVisibleTask$` but it has two issues : 
- It creates eager JS execution
- For some ms the content is not displayed, which creates some kind of "flash" effect.

**Comment by gioboa** [comment]
We moved this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.
